pycalphad/mapping/compat_api.py

```diff
@@ -0,0 +1,218 @@
+import numpy as np
+
+from pycalphad.mapping.primitives import STATEVARS
+from pycalphad.mapping.strategy import StepStrategy, BinaryStrategy, TernaryStrategy, IsoplethStrategy
+from pycalphad.mapping.plotting import plot_step, plot_binary, plot_ternary, plot_isopleth
+
+def binplot(database, components, phases, conditions, return_strategy=False, plot_kwargs=None, **map_kwargs):
+    """
+    Calculate the binary isobaric phase diagram.
+
+    Parameters
+    ----------
+    database : Database
+        Thermodynamic database containing the relevant parameters.
+    components : Sequence[str]
+        Names of components to consider in the calculation.
+    phases : Sequence[str]
+        Names of phases to consider in the calculation.
+    conditions : Mapping[v.StateVariable, Union[float, Tuple[float, float, float]]]
+        Maps StateVariables to values and/or iterables of values.
+        For binplot only one changing composition and one potential coordinate each is supported.
+    return_strategy : bool, optional
+        Return the BinaryStrategy object in addition to the Axes. Defaults to False.
+    map_kwargs : dict, optional
+        Additional keyword arguments to BinaryStrategy().
+    plot_kwargs : dict, optional
+        Keyword arguments to plot_binary().
+
+    Returns
+    -------
+    Axes
+        Matplotlib Axes of the phase diagram
+    (Axes, BinaryStrategy)
+        If return_strategy is True.
+
+    """
+    indep_comps = [key for key, value in conditions.items() if key not in STATEVARS and len(np.atleast_1d(value)) > 1]
+    indep_pots = [key for key, value in conditions.items() if key in STATEVARS and len(np.atleast_1d(value)) > 1]
```

Comment:
Are you referring to pycalphad.core.utils.get_state_variables? 

There's a lot of references in the mapping code to the hard-coded STATEVARS list which I think the final solution here would be to remove STATEVARS in favor of getting state variables from get_state_variables in case a user defines a custom state variable. An example this could benefit could be interfacial contribution for Gibbs-Thompson phase diagrams, in which a hard-coded STATEVARS list would not allow for.